fe/fe-core/src/main/java/com/starrocks/planner/IcebergMetadataScanNode.java

```diff
@@ -0,0 +1,136 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.starrocks.planner;
+
+import com.starrocks.analysis.TupleDescriptor;
+import com.starrocks.common.UserException;
+import com.starrocks.connector.RemoteMetaSplit;
+import com.starrocks.connector.SerializedMetaSpec;
+import com.starrocks.connector.metadata.LogicalIcebergMetadataTable;
+import com.starrocks.server.GlobalStateMgr;
+import com.starrocks.sql.plan.HDFSScanNodePredicates;
+import com.starrocks.thrift.THdfsScanNode;
+import com.starrocks.thrift.THdfsScanRange;
+import com.starrocks.thrift.TNetworkAddress;
+import com.starrocks.thrift.TPlanNode;
+import com.starrocks.thrift.TPlanNodeType;
+import com.starrocks.thrift.TScanRange;
+import com.starrocks.thrift.TScanRangeLocation;
+import com.starrocks.thrift.TScanRangeLocations;
+import org.apache.hadoop.shaded.com.google.common.base.Strings;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class IcebergMetadataScanNode extends ScanNode {
+    private static final Logger LOG = LogManager.getLogger(IcebergMetadataScanNode.class);
+    private final LogicalIcebergMetadataTable table;
+    private String icebergPredicate = "";
+
+    private final HDFSScanNodePredicates scanNodePredicates = new HDFSScanNodePredicates();
+    private final List<TScanRangeLocations> result = new ArrayList<>();
+    private String temporalClause;
+    private String serializedTable;
+
+
+    public IcebergMetadataScanNode(PlanNodeId id, TupleDescriptor desc, String planNodeName, String temporalClause) {
+        super(id, desc, planNodeName);
+        this.table = (LogicalIcebergMetadataTable) desc.getTable();
+        this.temporalClause = temporalClause;
+    }
+
+    public void preProcessIcebergPredicate(String icebergPredicate) {
+        this.icebergPredicate = icebergPredicate;
+    }
+
+    public HDFSScanNodePredicates getScanNodePredicates() {
+        return scanNodePredicates;
+    }
+
+    @Override
+    public List<TScanRangeLocations> getScanRangeLocations(long maxScanRangeLength) {
+        return result;
+    }
+
+    public void setupScanRangeLocations() throws UserException {
+        String catalogName = table.getCatalogName();
+        String originDbName = table.getOriginDb();
+        String originTableName = table.getOriginTable();
+
+        long snapshotId = -1;
+        // TODO(stephen): parse version by AstBuilder
+        if (!Strings.isNullOrEmpty(temporalClause)) {
+            String kw = "version as of";
+            temporalClause = temporalClause.substring(kw.length()).trim();
+            snapshotId = Long.parseLong(temporalClause);
+        }
+
+        SerializedMetaSpec SerializedMetaSpec = GlobalStateMgr.getCurrentState().getMetadataMgr()
```

Comment:
do you think it's better to reuse ` GlobalStateMgr.getCurrentState().getMetadataMgr().getRemoteFileInfos` which means that let's treat meatadata table as a connector